### ISSUE TITLE
Enhance error log message with destination blob is in an inconsistent state and --delete-destination-blob is not used

### ISSUE THREAD

**Issue by jolauMSFT** [Azure/azure-storage-azcopy#2578]
AzCopy team: 
Please enhance the error message when a copy fails due to the destination blob being in an inconsistent state. 
The current error message is not helpful enough to troubleshoot as it references the blob being invalid rather than inconsistent at the destination side. 
For the support case this led to hours of incorrect troubleshooting around the source blob and the copy commands. 
Eventual mitigation was to use --delete-destination-file flag 

Suggest the 400 error be enhanced from
The specified blob or block content is invalid.

to (for example):
A blob with specified name exists at <destination>. Potentially in an inconsistent state. Delete the blob at destination manually or try again with --delete-destination-file flag set.

### Which version of the AzCopy was used? 
10.23.0
### Which platform are you using? (ex: Windows, Mac, Linux)
Linux

##### Note: Please remove the SAS to avoid exposing your credentials. If you cannot remember the exact command, please retrieve it from the beginning of the log file.
azcopy copy "
<sourceFQDNRedacted>?ss=b&sig=REDACTED%3D&st=2020-02-1
2T19%3A38%3A20Z&se=2030-02-12T19%3A38%3A20Z&sv=2018-11-09&srt=sco&spr=https&sp=racwl&sr=sco" "
<destFQDNRedacted>?ss=b&sig=REDACTED%3D&st=2020-02-12T20%3A30%3A03Z&se=2030-02-12T20%3A30%3A03Z&sv=2018-11-09&srt=sco&spr=https&sp=racwl&sr=sco" --recursive --log-level WARNING &"

### What problem was encountered?
RESPONSE STATUS CODE ERROR
   PUT <destFQDNRedacted>?blockid=MDAwMDDQljZmo5JOxWNJMOO03XWvMDAwMDAwMDAwMDAwMjY4&comp=block&se=2030-02-12T20%3A30%3A03Z&sig=-REDACTED-&sp=racwl&spr=https&sr=
sco&srt=sco&ss=b&st=2020-02-12T20%3A30%3A03Z&sv=2018-11-09
   Accept: application/xml
   Content-Length: 0
   User-Agent: AzCopy/10.23.0 azsdk-go-service.Client/v1.2.0 (go1.19.12; linux)
   X-Ms-Client-Request-Id: 8ab018ee-c0d9-4cfc-43db-14cd14bcda32
   x-ms-copy-source: <sourceFQDNRedacted>?se=2030-02-12T19%3A38%3A20Z&sig=-REDACTED-&sp=racwl&spr=https&sr=sco&srt=sco&ss=b&st=2020-02-12T19%3A38%3A20Z&sv=20
18-11-09
   x-ms-source-range: bytes=2248146944-2256535551
   x-ms-version: 2023-08-03
   --------------------------------------------------------------------------------
   RESPONSE Status: 400 The specified blob or block content is invalid.
   Content-Length: 234
   Content-Type: application/xml
   Date: Thu, 15 Feb 2024 04:37:06 GMT
   Server: Windows-Azure-Blob/1.0 Microsoft-HTTPAPI/2.0
   X-Ms-Client-Request-Id: <See ICM 471659159>
   X-Ms-Error-Code: InvalidBlobOrBlock
   X-Ms-Request-Id: <See ICM 471659159>
   X-Ms-Version: 2023-08-03

### How can we reproduce the problem in the simplest way?
Create an uncommitted blob on destination storage account. Try to copy with the same destination blob name using azcopy.

### Have you found a mitigation/solution?
<See ICM 471659159>

**Comment by vibhansa-msft** [comment]
Hi,

In the ICM request-id shared in description was not the one causing this failure and intially all investigation went in analysis based on that request-id, which was in fact caused by some network issue in the bakend.
Once we realised this can not be root-cause of the failure, we looked at logs again and found the request-id mentioned is incorret. After correction course of investigation we figured out it's inconsistent state of blob at destination and suggested the mitigations.

For the error message part, AzCopy can only throw back what backend server returns. Here it just says "InvalidBlobOrBlock" and AzCopy as a tool has no idea why backend server returned this error code. It just translates this into appropriate message and throws it back to user. I do agree we need improvement here, but this has to be driven from backend rather than AzCopy. As this is a 4xx error from backend, AzCopy will just fail the copy job for that particular file.

Backend investigation also found that blob was in inconsistent state for last week, so deeper understanding on how that state was hit is also something worth investigating from your end.
